src/transformStateWithClones.js

```diff
@@ -7,7 +7,23 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let objToAddInArrResult = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.entries(action.extraData).forEach(([key, value]) => {
+        objToAddInArrResult[key] = value;
+      });
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(key => delete objToAddInArrResult[key]);
+    } else {
+      objToAddInArrResult = {};
+    }
```

Comment:
better to use `switch... case` here, also for adding could be used
Object.assign